File: 3_multiple_inheritance.py
# In Python, there's no direct concept of multiple inheritance.

class Animal:
    def __init__(self, name):
        self.name = name

    def eat(self):
        print(self.name, "can eat")

class Pet:
    def play(self):
        print("Playing with pet")

# Derived class inheriting from both Animal and Pet
class Cat(Animal, Pet):
    def __init__(self, name):
        super().__init__(name)

# Main function
def main():
    obj = Cat("Cat")
    obj.eat()   # Accessing eat() from Animal class
    obj.play()  # Accessing play() from Pet class

if __name__ == "__main__":
    main()

File: 3_multiple_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;

// Base class 1
class Animal {
public:
    std::string name;
    void eat() {
        std::cout << name << " Can eat" << std::endl;
    }
};

// Base class 2
class Pet {
public:
    void play() {
        std::cout << "Playing with pet" << std::endl;
    }
};

// Derived class inheriting from both Animal and Pet
class Cat : public Animal, public Pet {
public:
    Cat(std::string name) {
        this->name = name;
    }
};

int main() {
    Cat obj("Cat");
    obj.eat();  // Accessing eat() from Animal class
    obj.play(); // Accessing play() from Pet class
    
    return 0;
}
